### PR TITLE
Move to SQLAlchemy v2, add optimizations

### LINKED ISSUE #67: Migrate to SQLAlchemy 2.0
https://docs.sqlalchemy.org/en/20/changelog/migration_20.html

### PR DESCRIPTION
**Pull request by m-rtijn**
Closes #67

### CODE REVIEW

**Comment by m-rtijn** [comment]
Todo:
- evaluate index performance, check which are required
- see if some things can be changed into more efficient queries (possibly with extra indexes)

File: argostime/models.py [hunk 1]
```diff
@@ -116,30 +115,62 @@ def __str__(self):
...
+            return avg
         except statistics.StatisticsError:
             logging.debug("Called get_average_price for %s but no prices were found...", str(self))
             return -1
 
+    def get_average_price(self) -> float:
+        """Stub for new .average_price attribute"""
+        return self.average_price
+
+    def get_prices_since(self, since_time: datetime) -> list[Price]:
```
[the remark this answers is not in the document]
Comment (m-rtijn) [reply argostime/models.py:156]: I've just decided that we should be using python 3.10 then :P Will add an issue to fix any legacy type annotations.

**Comment by m-rtijn** [comment]
Page load times are significantly reduces now, this is quite some progress for #56

File: requirements_development.txt [hunk 2]
```diff
@@ -1,2 +1,5 @@
-mypy
-pylint
+-r requirements.txt
```
Comment (tweska) [inline requirements_development.txt:1]: What happened here? :^)

**Review by tweska — DISMISSED** [review]

**Pull request merged**